atom/common/api/atom_bindings.cc

```diff
@@ -203,6 +211,49 @@ v8::Local<v8::Value> AtomBindings::GetSystemMemoryInfo(v8::Isolate* isolate,
   return dict.GetHandle();
 }
 
+// static
+v8::Local<v8::Promise> AtomBindings::GetMemoryFootprint(v8::Isolate* isolate) {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate);
+  node::Environment* env = node::Environment::GetCurrent(isolate);
+  memory_instrumentation::MemoryInstrumentation::GetInstance()
+      ->RequestGlobalDumpForPid(base::GetCurrentProcId(),
+                                base::AdaptCallbackForRepeating(base::BindOnce(
+                                    &AtomBindings::DidReceiveMemoryDump,
+                                    base::Unretained(env), promise)));
+  return promise->GetHandle();
+}
+
+// static
+void AtomBindings::DidReceiveMemoryDump(
+    node::Environment* env,
+    scoped_refptr<util::Promise> promise,
+    bool success,
+    std::unique_ptr<memory_instrumentation::GlobalMemoryDump> global_dump) {
+  // Enter Node environment.
+  mate::Locker locker(env->isolate());
+  v8::HandleScope handle_scope(env->isolate());
+  v8::MicrotasksScope script_scope(env->isolate(),
+                                   v8::MicrotasksScope::kRunMicrotasks);
+  v8::Context::Scope context_scope(env->context());
+
+  if (success) {
+    bool resolved = false;
+    for (const memory_instrumentation::GlobalMemoryDump::ProcessDump& dump :
+         global_dump->process_dumps()) {
+      if (base::GetCurrentProcId() == dump.pid()) {
+        promise->Resolve(dump.os_dump().private_footprint_kb);
```

Comment:
it looks like os_dump()'s return has both private_footprint_kb and shared_footprint_kb. That's is pretty similar to the MemoryInfo API we were using before, which had privateBytes and sharedBytes.

Would it make sense to keep using MemoryInfo as a return value and store both values in it?

Reply:
Could this be used as a new backend for `app.getAppMetrics()`?

And if so -- _should_ it be?

Reply:
That's a good idea. 👍 
Do we want to keep the same api also, `getProcessMemoryInfo`? It will have to return a promise though.
What about working set size and peak working set size? The original API returned their values on all platforms, but those terms have meaning only on windows.

Reply:
Yes, I definitely want to add `resident_set_kb` and `shared_footprint_kb` to the same API. So. we can probably leave the same `getProcessMemoryInfo` API and return an updated `MemoryInfo` struct containing `{ resident_set_kb, shared_footprint_kb, private_footprint_kb}`.
`resident_set_kb` is mostly a better replacement for the existing working set size attributes.
Regarding requesting specific allocator dumps, I'm not sure if they should be part of this `process` API. They can be part of the `app.getAppMetrics` api though.